.github/workflows/CI-docker.yml

```diff
@@ -4,42 +4,40 @@ run-name: "Workflow performing CI step: push docker image to DockerHub"
 on:
   push:
     tags:
-      - v[0-9].[0-9].[0-9]*
+      - 'v[0-9].[0-9].[0-9]*'
   workflow_dispatch:
 
 env:
   RUST_BACKTRACE: 1
+  DOCKER_HUB_ORG: horizenlabs
+  DOCKER_IMAGE_NAME: nh-node
+  PROD_VERSION_REGEX: '^v[0-9]+\.[0-9]+\.[0-9]+(-[0-9]+)?(-rc[0-9]+)?$'
```

Comment:
I had a brief discussion with Luigi on this, I am not totally fine with this `PROD_VERSION_REGEX` logic.
Just to make sure we are aligned on what was our previous approach:

- `v1.0.0`: fine, tag `1.0.0` published, tag `latest` published
-` v1.0.0-rc.1`: fine, `1.0.0-rc.1` published, tag `latest` NOT published (because not an official/final version)
-` v1.0.0-beta.1`: fine, `1.0.0-beta.1` published, tag `latest` NOT published (as above)
- `v1.0.0-blablabla`: fine, `1.0.0-blablabla` published, tag `latest` NOT published (as above); maybe not super-good but acceptable
What I don't like very much is that with your new approach `v1.0.0-rc.1` would be published as `latest` (can be a problem for someone joining the network after the publishing of `v1.0.0-rc.1` but before the publishing of `v1.0.0`) and `v1.0.0-beta.1` would be rejected at all (but maybe we want to test it on some dedicated machines).

Reply:
If we are planning to drop `v` in the tag name, why not to drop it as well for github tag name as well? use 1.1.1 for github tag vs v1.1.1?

How about we define a fixed pattern for allowed tags to be published like ONLY
1) If 1.1.1 = PRODUCTION release tags are 1.1.1 and latest are published
2) If 1.1.1-rc[1-9] = DEV release and ONLY 1.1.1-rc[1-9] is published

Why do we need beta and smth else?
We can always add it if needed of course

Reply:
Let's drop the `v` in GH.
About the tagging patterns I like your proposal @ysibirski but I would also allow "free tags" like : 

If 1.1.1 = PRODUCTION release tags are 1.1.1 and latest are published can ONLY be done if tag is from MAIN branch
If 1.1.1-rc[1-9] = DEV release and ONLY 1.1.1-rc[1-9] is published can be done from ANY branch
If 1.2.0-anyword = only 1.2.0-anyword tag get published on DH. 

It can be useful for MC team to be able to push also specific experimental tags. 
We will anyway have the check on the maintainers keys.

Reply:
Ok we are perfectly aligned, thanks.